flocker/common/test/test_retry.py

```diff
@@ -191,6 +193,72 @@ def predicate():
             str(LoopExceeded(predicate, False)))
 
 
+class TimeoutTests(SynchronousTestCase):
+    """
+    Tests for :py:func:`timeout`.
+    """
+
+    def setUp(self):
+        """Initialize testing helper variables."""
+        self._deferred = Deferred()
+        self._timeout = 1.0
+        self._clock = Clock()
+
+    def _execute_timeout(self):
+        """Execute the timeout."""
+        timeout(self._clock, self._deferred, self._timeout)
+
+    def test_times_out(self):
+        """
+        Verify that a deferred that never fires is timed out at the correct
```

Comment:
Nit: no need for "Verify that" or "Assert that" in test docstrings. https://jml.io/test-docstrings/